backend/redis/conn_mgr.c

```diff
@@ -238,7 +238,307 @@ int dbBE_Redis_connection_mgr_conn_fail( dbBE_Redis_connection_mgr_t *conn_mgr,
+}
+
+static int
+dbBE_Redis_connection_mgr_create_connection_list( dbBE_Redis_connection_mgr_t *conn_mgr,
+                                                  dbBE_Redis_connection_t ***connections )
+{
+  // create a temporary list of connections where we remove valid master connections to see which links need to get fixed
+  *connections = (dbBE_Redis_connection_t**)calloc( conn_mgr->_connection_count, sizeof( dbBE_Redis_connection_t*) );
+  if( connections == NULL )
+    return -1;
+
+  int n,i;
+  for( n=0, i=0; DBBE_CONNECTIONS_TO_GO( conn_mgr, n, i ); ++n )
+  {
+    // skip empty slots
+    if( DBBE_CONNECTION_MGR_SLOT_EMPTY( conn_mgr, n ) )
+      continue;
+    // double check that it's not a broken connection
+    if(( conn_mgr->_broken[ n ] != NULL ) || ( conn_mgr->_connections[ n ] == NULL ))
+    {
+      LOG( DBG_ERR, stderr, "Inconsistent connection mgmt. Broken connection found in unbroken state\n" );
+      free( connections );
+      return -1;
+    }
+    (*connections)[ i++ ] = conn_mgr->_connections[ n ];
+  }
+
+  if( i != conn_mgr->_connection_count )
+  {
+    LOG( DBG_ERR,
+         stderr,
+         "Inconsistent connection mgmt. Number of tracked connections %d differs from active connections %d.\n",
+         conn_mgr->_connection_count,
+         i );
+    free( connections );
+    return -1;
+  }
+
+  return i;
+}
+
+static int
+dbBE_Redis_connection_mgr_compare_cluster_info( dbBE_Redis_cluster_info_t *new_cluster,
+                                                dbBE_Redis_cluster_info_t *old_cluster,
+                                                dbBE_Redis_connection_t **connections,
+                                                int connection_count )
+{
+  int c, n;
+  int master_count = 0;
+  for( c=0; c<dbBE_Redis_cluster_info_getsize( new_cluster ); ++c )
+  {
+    dbBE_Redis_server_info_t *si_new = dbBE_Redis_cluster_info_get_server( new_cluster, c );
+    if( si_new == NULL )
+      return -1;
+
+    char *url = dbBE_Redis_server_info_get_master( si_new );
+    if( url == NULL )
+      return -1;
+
+    dbBE_Redis_server_info_t *si_old = dbBE_Redis_cluster_info_get_server_by_addr( old_cluster, url );
+    if( si_old == NULL )
+      return -1;
+
+    // remove any master connection from the tmp_conns list so that only replicas are left
+    for( n=0; n<connection_count; ++n )
+    {
+      dbBE_Redis_connection_t *conn = connections[ n ];
+      if( conn == NULL ) continue; // skip any entries that we might have removed earlier
+      char *c_url = dbBE_Redis_connection_get_url( conn );
+      if( strncmp( c_url, url, DBR_SERVER_URL_MAX_LENGTH ) == 0 )
+      {
+        connections[ n ] = NULL;
+        ++master_count;
+        break; // we found a connection to the si_new master; check the next one now...
+      }
+    }
+  }
+  return master_count;
+}
+
+static int
+dbBE_Redis_connection_mgr_wipe_replica_connections( dbBE_Redis_connection_mgr_t *conn_mgr )
+{
+  int n, i;
```

Comment:
is `i` never used?

Reply:
it is used within this macro DBBE_CONNECTIONS_TO_GO()

Reply:
ok! Asking because I usually saw it incremented, while in this function is not

Reply:
oh yeah. you're right. It's probably a missing increment there. While the function would still do what it's supposed to do, it's unnecessarily iterating over unused space. I'll fix that.